pyro/contrib/epidemiology/compartmental.py

```diff
@@ -287,6 +314,8 @@ def fit(self, **options):
         mcmc = MCMC(kernel, **options)
         mcmc.run()
         self.samples = mcmc.get_samples()
+        if self.is_regional:
+            self.samples["auxiliary"] = self.samples["auxiliary"].unsqueeze(1)
```

Comment:
@neerajprad at some point you and I discussed properly aligning the particle dim of MCMC sample tensors to support replaying, but it looks like here we need to shape because the samples are simply `torch.stack()`ed. Do you remember any details of our conversation or our resolution?

Reply:
I believe our conversation was in the context of `Predictive` but looking at that logic now, I don't see how it can work with nontrivial event shape 😕 
https://github.com/pyro-ppl/pyro/blob/8633b7136946ab2ae2e16062503fe51c2aac8c38/pyro/infer/predictive.py#L58-L61

Reply:
OK @neerajprad I've refactored to use an `align_samples()` helper.